lib/plugins/aws/package/index.js

```diff
@@ -76,7 +80,11 @@ class AwsPackage {
       'aws:package:finalize:mergeCustomProviderResources': () => BbPromise.bind(this)
         .then(this.mergeCustomProviderResources),
 
+      'after:aws:package:finalize:mergeCustomProviderResources': () => BbPromise.bind(this)
+        .then(this.validateAgainstStackLimits),
+
       'aws:package:finalize:saveServiceState': () => BbPromise.bind(this)
+        .then(this.splitStack)
```

Comment:
splitstack could be exposed here as new lifecycle event that can be hooked. Semantically it has nothing to do with "saveServiceState".
`aws:package:finalize:splitStack` should be the new event exposed, that does the splitStack operation.